modules/gdscript/gdscript_editor.cpp

```diff
@@ -1209,6 +1294,16 @@ static void _find_identifiers_in_base(const GDScriptCompletionIdentifier &p_base
 						if (!base_type.is_meta_type) {
 							List<PropertyInfo> members;
 							scr->get_script_property_list(&members);
+
+							HashSet<StringName> deprecated_properties;
+							if (base_type.class_type && class_doc_map.has(base_type.class_type->get_global_name())) {
```

Comment:
I think you shouldn't use `get_global_name()` to determine the doc class name. Because for unnamed classes, inner classes and autoload singletons the name may differ from the global name.

After #91060 there is a new method `Script::get_doc_class_name()`, but there is no counterpart for `GDScriptParser::ClassNode`. I reused `GDScriptDocGen::doctype_from_gdtype()` (which uses `GDScriptDocGen::_get_class_name()`).

Reply:
Thanks! I'll look more into this once I have the compatibility method stuff working and CI tests passing.